File: JAVA/PartitionArrayAccordingToGivenPivot.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

public class PartitionArrayAccordingToGivenPivot {

    public static void main(String[] args) {
        System.out.printf("Resultado: %d\n", pivotArray( new int[]{3,2,2}, 1 ));
    }

    public static int[] pivotArray(int[] nums, int pivot) {
        int[] res = new int[nums.length];
        int k = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i] < pivot){
                res[k++] = nums[i];
            }
        }
        
        for(int i=0;i<nums.length;i++){
            if(nums[i] == pivot){
                res[k++] = nums[i];
            }
        }
        
        for(int i=0;i<nums.length;i++){
            if(nums[i] > pivot){
                res[k++] = nums[i];
            }
        }
        
        return res;
    }

}

File: GO/PartitionArrayAccordingToGivenPivot.go
/**
 * @version GO 1.26.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func pivotArray(nums []int, pivot int) []int {
	res := make([]int, len(nums))

	k := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] < pivot {
			res[k] = nums[i]
			k++
		}
	}
	for i := 0; i < len(nums); i++ {
		if nums[i] == pivot {
			res[k] = nums[i]
			k++
		}
	}
	for i := 0; i < len(nums); i++ {
		if nums[i] > pivot {
			res[k] = nums[i]
			k++
		}
	}
	return res
}
